hack/gcr-push.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+# Return value is written into LATEST
+LATEST=""
+function latest_release() {
+# Loop through the tags since pre-release versions come before the actual versions.
+# Iterate til we find the first non-pre-release
+    for t in $(git tag -l --sort=-v:refname);
+    do
+        # If the tag has alpha, beta or rc in it, it's not "latest"
+        if [[ "$t" == *"beta"* || "$t" == *"alpha"* || "$t" == *"rc"* ]]; then
+            continue
+        fi
+        LATEST="$t"
+        break
+    done
+}
+
+
+# Always publish for master
+if [ "$BRANCH" == "master" ]; then
+    VERSION="$BRANCH" make all-containers all-push
+fi
+
+# Publish when TRAVIS_TAG is defined.
+if [ ! -z "$TRAVIS_TAG" ]; then
```

Comment:
According to the [Travis environment variables](https://docs.travis-ci.com/user/environment-variables/), this should be set to the name of the tag that triggered the build, which should also be the one we want to push images for, right? Can we not just use that directly?

Reply:
By directly, do you mean within the Makefile? I'd like to avoid that so our Makefile isn't linked to the provider.